File: leetcode/medium/1060-missing-element-in-sorted-array/1060-missing-element-in-sorted-array.py
from typing import List


class Solution:
    def missingElement(self, nums: List[int], k: int) -> int:
        target = nums[0]
        index = 0
        count = 0
        length = len(nums)

        while (count < k):
            if index == length:
                return target + k - count - 1

            if target < nums[index]:
                target += 1
                count += 1
            else:
                index += 1
                target += 1

        return target - 1

File: leetcode/medium/1060-missing-element-in-sorted-array/1060-missing-element-in-sorted-array.kt
class Solution {
    fun missingElement(nums: IntArray, k: Int): Int {
        var target = nums[0]
        var index = 0
        var count = 0
        var length = 0

        while (count < k) {
            if (index == nums.size) {
                return target + k - count - 1
            }

            if (nums[index] == target) {
                index++
            } else {
                count++
            }
            target++
        }

        return target - 1
    }
}
